src/Symfony/Component/Routing/Requirement.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing;
+
+/**
+ * A collection of universal regular-expression constants to use as route parameter requirements.
+ */
+final class Requirement
```

Comment:
instead of `final class`, we could write `enum` here, this would make the class not instantiable also, wdyt?

Reply:
I don't know much about enums yet but it seems they don't fit right now. An enum cannot implement `__toString()` and `->value` cannot be used as a constant expression directly. So using `Requirement::UidBase58` or `Requirement::UidBase58->value` does not work.

We could however accept `\BackedEnum` in the `Route $requirements` array and get their value at this moment, but what's the real benefit vs constants?

To make the class not instantiable I can just add a private constructor.

Reply:
I'm saying we shouldn't keep the constants ;)